Renamed JS and have it make use of intent

NSoiffer opened the pull request:
This updates the polyfill and makes it clear this is not about mtable (attributes) in general.

I tried to add some CSS to move the label, but I didn't get it right.

davidcarlisle commented on mlabeledtr/mlabeledtr.js:
```diff
@@ -51,24 +51,33 @@ function handleLabeledRows(mtable) {
+    if (row.tagName === 'mlabeledtr') {
+      // move the label to the left or right side of a new "mtr" (instead of "mlabeledtr")
+      let label = row.firstElementChild;
+      label.setAttribute('intent', ':equation-label');
```
@dginev  it is not doing inference here, the input markup mlabeledtr/mtd is being replaced by mtr/mtd intent=:equation-label` So the input has specified this is an equation label

dginev replied:
As long as this is MathML 3 or earlier, that's fine. My suggestion would have only made sense in a MathML 4 context, but the element is removed there so not applicable.

davidcarlisle commented on mlabeledtr/mlabeledtr.js:
```diff
@@ -51,24 +51,33 @@ function handleLabeledRows(mtable) {
+        newRow.setAttribute(attr.name, attr.value);
+      }
+      // leave the label as the first element or move it to the right (last element)
+      let mtd = row.children[side=='left' ? 0 : 1];
```
mathml4 has removed the entire mlabeledtr element so the input is mathml3 (or mathml-4 legacy schema, that is the point of the polyfill to fill in mathml3 (or 4) features not in core.

dginev replied:
I see. That is a bit confusing to me.

It would have been nice to have some marker for polyfills that only apply to versions 3 and earlier, and the "active spec" polyfills made to handle MathML 4 Full.

As one example, polyfills meant to handle MathML 4 Full which do anything with the `intent` attribute should be checking if that attribute was already set in the input markup and preserve/merge that information. As an made up example, 
```xml
<menclose notation="updiagonalstrike" intent="wrong-approach($expr)"><mrow arg="expr">...
```

Shouldn't get `intent="cancel($expr)"` as an override when polyfilling. 

I was thinking of `<mlabeledtr intent="...">` here, but that isn't a MathML 3 use.

davidcarlisle replied:
> As one example, polyfills meant to handle MathML 4 Full which do anything with the intent attribute should be checking if that attribute was already set in the input markup and preserve/merge that information.

yes agree with that

NSoiffer commented:
I added `:no-equation-label`. That was easy. It turns out dealing with an existing `intent` was less easy (about 15 lines of code).

The CSS problem I had was because I had a typo in the CSS file name. That's fixed. I left in the CSS file reference awaiting Deyan's feedback on dropping it.

I added tests in the HTML page that write to the console if the intent is not as expected in the test cases. Two things for review:
* did I miss a case that should be tested
* the JS for the test needed to run after the other JS ran. I didn't know a good way to do that and feel back on a 100ms sleep. If one of you knows of the right way to make sure the script runs later, please fix it and/or let me know.

I did _not_ write code to write to the console for the left/right side mapping. I thought visual inspection is good enough.

davidcarlisle commented on mlabeledtr/mlabeledtr.js:
```diff
@@ -47,34 +47,67 @@ function handleLabeledRows(mtable) {
+    return;
+  }
+  let intentValue = mtd.getAttribute('intent');
+  let iOpenParen = intentValue.indexOf('(');
```
while in general polyfills should avoid over-writing existing intent, I'm not sure this is needed here, any existing use (should be) mathml3 so not having intent attributes. there shouldn't be intent attributes on the first child of an mlabeledr so it should safe to just insert intent="@equation-label`

NSoiffer commented:
@dginev: any comments about dropping the CSS and dropping the check for intent?

NSoiffer commented:
Do you have a way to limit the CSS to just `mtables` with `mlabeledtr` rows? Or do you have a suggested name for css?

davidcarlisle commented:
> Do you have a way to limit the CSS to just `mtables` with `mlabeledtr` rows? Or do you have a suggested name for css?

the css only really works for one table, it has fixed widths and fixed column numbers (and i never found a way to make a generic version) if using that version i would just inline it into a `<style>` element so it doesn't easily leake out to other documents.

The polyfill should just concern itself with fe-writing `<mlabeledtr` to `<mtr` / `<mtd intent=":equation-lablel`   Getting CSS so the latter works is an issue but is an issue whether or not the polyfill from `<mlabeledtr` is used, so should be separate.

dginev commented:
> Do you have a way to limit the CSS to just mtables with mlabeledtr rows? 

That at least is easy:

We could have the polyfill add a `mlabeledtr-polyfill` class on the `mtable` then use a `mtable.mlabeledtr-polyfill` selector.

davidcarlisle commented:
> > Do you have a way to limit the CSS to just mtables with mlabeledtr rows?
> 
> That at least is easy:
> 
> We could have the polyfill add a `mlabeledtr-polyfill` class on the `mtable` then use a `mtable.mlabeledtr-polyfill` selector.

the css should apply whether or not the polyfill is used so perhaps use 

`mtable[intent=":system-of-equations"]`

to select the relevant `mtable`

dginev commented:
@NSoiffer I think @davidcarlisle is happy to shepherd the PR. Let me know via email if you need anything from me.

davidcarlisle commented:
my above comment re css was misguided, the css is just referenced by the example html and not injected by any use of the polyfill so that's fine,
